docs/ldml/tr35-personNames.md

```diff
@@ -746,35 +755,43 @@ The following process is used to produce initials when they are not supplied by
 
-2. If a locale **doesn’t** require spaces between words, there are two cases, based on whether the foreign name is written in the locale’s script, or the foreign name is left in its native script. In both cases, patterns from the **locale of the name** are used. For example, the formatting locale might be Japanese, and the locale of the PersonName object might be de_CH, German (Switzerland), such as Albert Einstein.
+2. If a locale **doesn’t** require spaces between words, there are two cases, based on whether the name is foreign or not (based on the PersonName objects explicit or calculated locale's language subtag). For example, the formatting locale might be Japanese, and the locale of the PersonName object might be de_CH, German (Switzerland), such as Albert Einstein.
 
-    1. **The foreign name is written in the formatting locale’s script.** In that case, the **foreignSpaceReplacement** is substituted for each space in the patterns from the _locale of the name_. Here are examples for Albert Einstein in Japanese and Chinese:
+    1. **The name is native** In that case, the patterns are used as is.
+    2. **The name is foreign** In that case, the **foreignSpaceReplacement** is substituted for each space formatted name. Here are examples for Albert Einstein in Japanese and Chinese:
         * [アルベルト<span style="background-color:aqua">・</span>アインシュタイン](https://ja.wikipedia.org/wiki/%E3%82%A2%E3%83%AB%E3%83%99%E3%83%AB%E3%83%88%E3%83%BB%E3%82%A2%E3%82%A4%E3%83%B3%E3%82%B7%E3%83%A5%E3%82%BF%E3%82%A4%E3%83%B3) 
         * [阿尔伯特<span style="background-color:aqua">·</span>爱因斯坦](https://zh.wikipedia.org/wiki/%E9%98%BF%E5%B0%94%E4%BC%AF%E7%89%B9%C2%B7%E7%88%B1%E5%9B%A0%E6%96%AF%E5%9D%A6) 
-    2. **The foreign name is written in a different script.** In that case, the patterns from the **locale of the name** are used as is.
-        * [Albert Einstein](https://de.wikipedia.org/wiki/Albert_Einstein) 
+
 
 In both cases, the ordering may be changed according to the **Name Order for Locales** settings that each locale provides. If the PersonName object does not supply a locale for a name, then a default locale will be derived based on other information (such as the script of the characters in the name fields).
 
-> **Note** In the tech preview, the structure isn't yet powerful enough to handle cases with `foreignSpaceReplacement` where the formatting locale doesn’t need spaces between words, but the name locale has the same ordering as the formatting locale. 
-> For example, consider where the formatting locale is Thai, and the name is in English, but transliterated into Thai.
+Remember that **a name in a different script** will use a different locale for formatting. 
+For example, when formatting a name for Japanese, if the name is in the Latin script, a Latin based locale will be used to format it, such as when “Albert Einstein” appears in Latin characters on [Albert Einstein](https://ja.wikipedia.org/wiki/Albert_Einstein) 
```

Comment:
See my comments above.  I think this text needs to be incorporated into the description above, and I don't think you can separate out the separate-script case as cleanly as you're doing here.

Reply:
I agree that it needs to be reworked. Here's my trial.

1. If the name locale's script is different than the formatter locale's script, then use a formatter for the name locale. So when a Hindi formatter hits a name in Cyrillic, the code creates and uses a Cyrillic formatter to format that name. So we can assume for the rest of this discussion that the Hindi formatter will only need to handle names in Hindi script.

2. There are then two characteristics that need to be established.
* what is the nameOrder? — affects which patterns are chosen
* what is the spaceReplacement? —  affects post-processing of the composed string

3. The nameOrder is based on a lookup of the name locale in the name order data in the formatting locale the. (It isn't a simple lookup, because we follow the parent chain until there is a map.

4. The spaceReplacement is a choice between the nativeSpaceReplacement and the foreignSpaceReplacement, also based on the locale of the formatter and locale of the name. Most locales have " " = nativeSpaceReplacement = foreignSpaceReplacement, so they are easy.

* The foreignSpaceReplacement is provided by data in the locale.
*The nativeSpaceReplacement is currently calculated, based on the LineBreak information for the script. It is either " " or "".

The choice of which to use for the spaceReplacement is calculated: if the language subtag of the formatter's locale equals the language subtag of the name's locale, then the nativeSpaceReplacement is used, otherwise the foreignSpaceReplacement.
 
(Ideally we should add some more structure to make spaceReplacement more data-driven — filed https://unicode-org.atlassian.net/browse/CLDR-16457)

Reply:
This isn't what I have implemented on the ICU side, and I think it bears some discussion.  I also don't think step 1 is quite right, because it doesn't draw a distinction between the name **locale's** script and the **name's** script.  What we originally had was that we'd use a different formatter if the name locale had a different script **AND** and name itself was using that script (or something other than the formatter locale's script).  If you had a foreign name that had been transliterated into the native script, you kept the current formatter and used the foreignSpaceReplacement.  And is the CLDR code doing this at all right now?  I didn't see anything, but maybe I missed it.  I also think we need more test cases around this, on both the ICU and CLDR sides.